src/controllers/transfer/transfer.ts

```diff
@@ -289,6 +366,62 @@ export class TransferController extends EventEmitter {
     this.checkIsRecipientAddressUnknown()
   }
 
+  #setAmount(fieldValue: string) {
+    if (!fieldValue) {
+      this.amount = ''
+      this.amountInFiat = ''
+      return
+    }
+
+    const tokenPrice = this.selectedToken?.priceIn.find(
+      (p) => p.baseCurrency === HARD_CODED_CURRENCY
+    )?.price
+
+    if (!tokenPrice) {
+      this.amount = fieldValue
+      this.amountInFiat = ''
+      return
+    }
+
+    if (this.amountFieldMode === 'fiat' && this.selectedToken?.decimals) {
+      this.amountInFiat = fieldValue
+
+      // Get the number of decimals
+      const amountInFiatDecimals = fieldValue.split('.')[1]?.length || 0
+      const { tokenPriceBigInt, tokenPriceDecimals } = convertTokenPriceToBigInt(tokenPrice)
+
+      // Convert the numbers to big int
+      const amountInFiatBigInt = parseUnits(fieldValue, amountInFiatDecimals)
+      const precision = 16
+      const pow = BigInt(10 ** precision)
+
+      this.amount = formatUnits(
+        (amountInFiatBigInt * pow) / tokenPriceBigInt,
+        // Don't lose precision
+        amountInFiatDecimals + precision - tokenPriceDecimals
```

Comment:
Actually, it's not very likely, but it's possible to lose precision if the division results in more than 16 decimals, right?

But as I said, it's not something that will happen in 99% of cases.